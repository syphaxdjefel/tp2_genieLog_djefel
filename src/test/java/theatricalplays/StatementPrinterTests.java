package theatricalplays;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import java.util.List;
import java.util.Map;

import static org.approvaltests.Approvals.verify ;
import static org.approvaltests.Approvals.verifyHtml;


public class StatementPrinterTests {

   @Test
    void exampleStatement() {
        Map<String, Play> plays = Map.of(
                "hamlet",  new Play("Hamlet", PlayType.TRAGEDY),
                "as-like", new Play("As You Like It", PlayType.COMEDY),
                "othello", new Play("Othello", PlayType.TRAGEDY));

               
    Customer customer = new Customer("BigCo", "12345", 160);


     Invoice invoice = new Invoice(customer, List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)));
        Statement statement = new Statement(invoice, plays);

    // Obtenir la repr√©sentation en texte
    String result = statement.renderAsHtml();

        verifyHtml(result);
    }

    
}
