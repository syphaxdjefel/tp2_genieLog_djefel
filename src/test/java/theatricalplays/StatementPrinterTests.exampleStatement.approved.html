<html><head><title>Statement</title></head><body><h1>Statement for BigCo</h1><table><thead><tr><th>Play</th><th>Seats</th><th>Cost</th></tr></thead><tbody><tr><td>Hamlet</td><td>55</td><td>$650.00</td></tr><tr><td>As You Like It</td><td>35</td><td>$580.00</td></tr><tr><td>Othello</td><td>40</td><td>$500.00</td></tr></tbody></table><p>Amount owed is $1,730.00</p><p>You earned 47 credits</p></body></html>